clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1799,4 +1801,42 @@ void handleFailure(Throwable throwable) {
 
         return new DescribeReplicaLogDirResult(new HashMap<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>>(futures));
     }
+
+    public CreatePartitionsResult createPartitions(Map<String, NewPartition> newPartitions, final CreatePartitionsOptions options) {
+        final Map<String, KafkaFutureImpl<Void>> futures = new HashMap<>(newPartitions.size());
+        for (String topic: newPartitions.keySet()) {
+            futures.put(topic, new KafkaFutureImpl<Void>());
+        }
+        final Map<String, NewPartition> requestMap = new HashMap<>(newPartitions);
```

Comment:
Can newPartitions be passed directly (without creating requestMap) ?

Reply:
No, because the `newPartitions` is the one passed by the caller who could modify it while it's being concurrently accessed by the `Call`.